src/catacharset.cpp

```diff
@@ -415,21 +415,31 @@ std::u32string utf8_to_utf32( const std::string_view str )
 std::vector<std::string> utf8_display_split( const std::string &s )
```

Comment:
When/if this change is merged, I think we should just remove this older `utf8_display_split` function and replace all usages with the new fancy `utf8_display_split_into` that you did here. All other usages in `bodygraph.cpp`, `main_menu.cpp` and `veh_type.cpp` seem like they could just as well use this better `utf8_display_split_into` and have better performance by using it. If it's ok with you, I'll make a note of making a PR of doing those changes once this is merged. Or if you'd rather do that yourself, or maybe had that planned already. I'm happy to help if you think that's a good idea. Anyway, these changes look really nice! :+1:

Reply:
They could but they also aren't showing up nearly anywhere as hard as map gen was. Single largest allocation count by stack in the whole load sequence, by a large factor.

Reply:
Happy for you to do it.

Reply:
We can also change this function to return a vector of string view.
The one-liner
```cpp
std::vector<std::string_view> chars = utf8_display_split(str);
```
is more concise to read than
```cpp
std::vector<std::string_view> chars;
utf8_display_split_into(str, chars);
```
in non extremely performance sensitive use cases where persisting the allocated vector buffer is not needed.

Reply:
Yeah, that would be everywhere else, where I didn't want to do it lol. But sure I can do it here.

Reply:
Yeah I mean for future pull requests.